### PR TITLE
Feature/remove service users

### PR DESCRIPTION
**Pull request by eguzki**
Method to delete all users from a service. Implement method to call Backend internal endpoint:
```
DELETE /internal/services/#{service_id}/users
```

### CODE REVIEW

**Review by miguelsorianod — APPROVED** [review]

File: lib/3scale/core/errors.rb
Comment (davidor) [inline lib/3scale/core/errors.rb]: it's already defined a few lines above: https://github.com/3scale/pisoni/pull/15/commits/a26d89d86295cabcac6370d4c4759bb86b577f0f#diff-a761ded015e35c580d7119a41d7ed7c2R64

File: lib/3scale/core/users.rb
Comment (davidor) [inline lib/3scale/core/users.rb]: Shouldn't this be in the existing `User` class?

File: spec/users_spec.rb
Comment (davidor) [inline spec/users_spec.rb]: With this, we can't really be sure about whether the users were deleted.

I would create some and then verify that they are no longer there. We have methods for that in the `User` class.

**Comment by eguzki** [comment]
Tests failing, but nothing to do with this PR. Fixed in https://github.com/3scale/pisoni/pull/16

File: lib/3scale/core/errors.rb
Comment (eguzki) [reply lib/3scale/core/errors.rb]: That's for `User` class. We are defining for `Users` class.

File: lib/3scale/core/users.rb
Comment (eguzki) [reply lib/3scale/core/users.rb]: `User` contains `username` and other attributes. Could be implemented here, but I think it is cleaner with a new class definition

File: spec/users_spec.rb
Comment (eguzki) [reply spec/users_spec.rb]: I will implement. But I consider tests in pisoni should not check about functionality and logic from apisonator.

Comment (eguzki) [reply spec/users_spec.rb]: In apisonator there should be tests to check that https://github.com/3scale/apisonator/pull/38

File: lib/3scale/core/users.rb
Comment (davidor) [reply lib/3scale/core/users.rb]: I think that having a single class is more common. You could give the method a different name to be more explicit like `User.delete_all_for_service(service_id)`.

Comment (eguzki) [reply lib/3scale/core/users.rb]: @unleashed your call

Comment (unleashed) [reply lib/3scale/core/users.rb]: I also think this belongs in `User`. Having a class for this with the plural form is confusing to me (I also might be spoiled by Rails model conventions).

**Comment by eguzki** [comment]
Implemented in `User` class. Could not reuse `User` class methods like `base_uri` or `check_params` because the context does not apply for `Delete Users` operation, there is no single `username` implied as required by the methods.

@davidor review plz

**Comment by davidor** [comment]
Looks good @eguzki ,  please squash those "fixup" commits before merging.

**Review by davidor — APPROVED** [review]

**Pull request merged**
